src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java

```diff
@@ -1371,4 +1370,38 @@ public void testPlayerSpawnParticle_Incorrect_DataType()
 		});
 	}
 
+	@Test
+	public void testDisconnect()
+	{
+		assertTrue(player.isOnline());
+		assertTrue(player.disconnect());
+		assertFalse(player.isOnline());
+		assertFalse(server.getOnlinePlayers().contains(player));
+		server.getPluginManager().assertEventFired(PlayerQuitEvent.class);
+	}
+
+	@Test
+	public void testReconnect()
+	{
+		if (player.isOnline())
+		{
+			player.disconnect();
+		}
+
+		assertFalse(player.isOnline());
+		assertTrue(player.reconnect());
+		assertTrue(player.isOnline());
+		assertTrue(server.getOnlinePlayers().contains(player));
+		assertTrue(player.hasPlayedBefore());
+		server.getPluginManager().assertEventFired(PlayerJoinEvent.class);
+	}
+
+	@Test
+	public void testReconnectWithoutJoiningBefore()
```

Comment:
```suggestion
	void reconnect_HasntJoinedBefore_ThrowsException()
```

Reply:
We should probably establish Naming Conventions for Tests first